Group 1 - Final Project - Final Pull Request

rcsawka opened the pull request:
Group 1 Proposal Pull Request

rcsawka committed:
Create proposal.md

rcsawka committed:
copied from drive

rcsawka committed:
formatting

rcsawka committed:
formatting

Commit:
Starter code for server and react project

charlieroberts commented:
Hi, I love the idea for this project, and I don't really know of anything similar... it's almost like personal ads for academic pursuits, which I think is fantastic. If I read between the lines correctly, it also sounds like some of you are really interested in seeing this happen, which is great.

That said, it's a big project! Please think a bit about what a baseline would be for you to complete the project successfully, and if at all possible build up from there to the final specifications. 

I think Mikel (the TA) has some experience using OAuth with the WPI login system, which seems like it might be ideal for this. Reach out to him and see if he can provide help; if not I can point you to another student who has made extensive use of this in work he does on campus.

Looking forward to seeing how far you get with this!

ChrisVieira committed:
Added skeleton files to main branch of project

Moved starter files in a separate folder so they aren't confused with files developers are working on

ChrisVieira committed:
Important change, correct link now used

Commit:
oauth branch

Commit:
added css links login

Commit:
Submit functionality working

ChrisVieira committed:
Changes to styling the dashboard page

The dashboard now has a grid styling method with a fixed header. Most items have colors matching the style of the webpage and the appropriate borders.

Commit:
all routing done

ChrisVieira committed:
Added styling for the login page

Some server changes to change the send file action to be a function call using fs. Added some libraries that are commonly used in other areas.

ChrisVieira committed:
Merge pull request #1 from rcsawka/newdashboard

Newdashboard

ChrisVieira committed:
Merge branch 'main' into oauth

rcsawka committed:
Merge pull request #2 from rcsawka/oauth

Merging oauth into main

ChrisVieira committed:
Making the user.id function properly and cleaning the code a bit after merging

ChrisVieira committed:
Merge pull request #3 from rcsawka/oauth

Making the user.id function properly and cleaning the code a bit afte…

ChrisVieira committed:
Delete .env

Removing sensitive info

ChrisVieira committed:
change to ensure the .env file is ignored

Commit:
Filter functionality complete with reset button

ChrisVieira committed:
refactoring server, working on adding profile documents

ChrisVieira committed:
styling dashboard and login, switched discord image

ChrisVieira committed:
Merge pull request #4 from rcsawka/filtering

Filtering

rcsawka committed:
Update README.md

rcsawka committed:
Update README.md

ElaineChenWPI committed:
updated profile

rcsawka committed:
Update README.md

rcsawka committed:
Update README.md

rcsawka committed:
Update README.md

aburke921 committed:
mimicked dashboard layout for profile

rcsawka committed:
Update README.md

rcsawka committed:
Update README.md

rcsawka committed:
Update README.md

rcsawka committed:
Update README.md

rcsawka committed:
Update README.md

aburke921 committed:
added phone number with formatting to profile

aburke921 committed:
Most of Profile data passed to server

aburke921 committed:
Can now pass profile Phone Number to server

rcsawka committed:
Update README.md

aburke921 committed:
Default profile Image set

when the image is clicked on, file explorer opens

Commit:
Update README.md

aburke921 committed:
updated profileIMG

aburke921 committed:
Merge branch 'profile_combined' into maintain_profileid

aburke921 committed:
Merge pull request #5 from rcsawka/maintain_profileid

merging maintain_profileid (chris) into profile_combined (ashley)

rcsawka committed:
Update README.md

aburke921 committed:
Adds courses and skills to their relatiosns when user creates profile

aburke921 committed:
Adds & Edits profile properly

aburke921 committed:
Added delete icon and callback function

does not yet delete, only prints to console the profileID of the user that created it

rcsawka committed:
Update README.md

ChrisVieira committed:
Quick bug fixes

Made profile id process consistent between methods, added packages to npm install.

ChrisVieira committed:
Styling fix for header bar with placeholder text

rcsawka committed:
Update README.md

aburke921 committed:
Merge pull request #6 from rcsawka/regular-profile

Regular profile merged into main

rcsawka committed:
Update README.md

ChrisVieira committed:
Big fix

Fixed dashboard error where skills and languages can be null

ChrisVieira committed:
Merge branch 'main' of https://github.com/rcsawka/final_project

Commit:
Inserting class and skill relations

Commit:
Update README.md

Commit:
Change format of profile insert

ChrisVieira committed:
Redirect functionality working

Commit:
Remove default for select

Commit:
Sending profileID as number and formatting studentclassrelation documents

Commit:
Displaying skills for profile correcly in mongo

ChrisVieira committed:
Profile page populated with existing profile

ChrisVieira committed:
Merge pull request #7 from rcsawka/populate_profile_frontend

Profile page populated with existing profile

ChrisVieira committed:
Removed print statements

Commit:
Update README.md

ChrisVieira committed:
Big fixes and working on profile persistence

Called set state after submit action, this doesn't seem to be triggering. Sets list attributes to empty lists so they aren't undefined. Cast profileID to number for consistency.

Commit:
Querying all the information needed to populate profile page

aburke921 committed:
delete post now working

Commit:
Populating profile almost working. Removed profile picture from profile page

Commit:
Update README.md

ChrisVieira committed:
Adding consistency to profile population

ChrisVieira committed:
Merge pull request #8 from rcsawka/redirects

Redirect functionality working

ChrisVieira committed:
Merge branch 'main' into populate_profile

ChrisVieira committed:
Merge pull request #9 from rcsawka/populate_profile

Populate profile

ChrisVieira committed:
Cast profilid to number for redirect to work

aburke921 committed:
Merge branch 'main' into delete_posts_real_branch

aburke921 committed:
Merge pull request #10 from rcsawka/delete_posts_real_branch

delete post now working

Commit:
got delete working

Commit:
100 login lighthouse

rcsawka committed:
Update README.md

rcsawka committed:
Update README.md

ChrisVieira committed:
changed package.json file to ensure it can start on other services

rcsawka committed:
Update README.md

Commit:
Adding description to select labels

rcsawka committed:
Update README.md

rcsawka committed:
Update README.md

aburke921 committed:
Changed mouse to non-clickable over Outlook Login option

aburke921 committed:
Merge branch 'main' of https://github.com/rcsawka/final_project

ChrisVieira committed:
changed profileID to go off of requests

ChrisVieira committed:
Small change to readme to update most recent tasks

rcsawka committed:
Update README.md

ChrisVieira committed:
Small typo fix in readme

ChrisVieira committed:
Merge branch 'main' of https://github.com/rcsawka/final_project into main

rcsawka commented:
Here is our final pull request with our functional app and repository

rcsawka commented:
Here is our pull request with our repository and our functional app

rcsawka commented:
Here is our final pull request with our repository and functional application!
